src/settings.rs

```diff
@@ -26,7 +30,194 @@ impl Settings {
 }
 
 impl RepositorySettings {
-    pub fn closed_since_date(&self) -> DateTime<Utc> {
-        Utc::now()
+    pub fn closed_since_date(&self) -> Option<DateTime<Utc>> {
+        if let Some(ref s) = self.since {
+            match *s {
+                Since { quantity: q @ _, unit: SinceSpan::Day } => {
+                    return Some(Utc::now().checked_sub_signed(Duration::days(q as i64)).unwrap())
+                },
+                Since { quantity: q @ _, unit: SinceSpan::Week } => {
+                    return Some(Utc::now().checked_sub_signed(Duration::weeks(q as i64)).unwrap())
+                },
+            }
+        };
+
+        None
+    }
+}
+
+#[derive(Debug, PartialEq)]
+pub struct Since {
+    quantity: u64,
+    unit: SinceSpan,
+}
+
+impl FromStr for Since {
+    type Err = Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let tokens: Vec<&str> = s.split_whitespace().collect();
+        if tokens.len() != 2 {
+            return Err(format_err!("Value for 'since' should be exactly two words (found {} words in '{}')", tokens.len(), s));
+        }
+
+        let quantity = tokens[0].parse::<u64>()?;
+        let interval = tokens[1].parse::<SinceSpan>()?;
+
+        Ok(Since { quantity: quantity, unit: interval, })
     }
 }
+
+impl<'de> Deserialize<'de> for Since {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+        where D: Deserializer<'de>
+    {
+        let val = String::deserialize(deserializer)?;
+        val.parse::<Since>().map_err(serde::de::Error::custom)
+    }
+}
+
+#[derive(Debug, PartialEq)]
+pub enum SinceSpan {
+    Day,
+    Week,
+}
+
+impl FromStr for SinceSpan {
+    type Err = Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s.to_lowercase().as_str() {
+            "day"  | "days"  => Ok(SinceSpan::Day),
+            "week" | "weeks" => Ok(SinceSpan::Week),
+            _ => Err(format_err!("'{}' is not a recognized interval", s)),
```

Comment:
On the time front, do we want to add months as well?

Reply:
I thought about doing that at first, however the problem is that date math involving months is...not really well defined. For example, what is "`March 31st - 1 month`"?

Reply:
ah, fair